### PR TITLE
feat: Add `engine` and `engine_version_actual` outputs

### PR DESCRIPTION
**Pull request by stefan-matic**
## Description

Added new outputs for `engine` and `engine_version_actual` in module, examples, and README.

## Motivation and Context

The output value for `engine` is needed when creating Secrets Manager RDS Credentials in the same way how it's created via the console. The keys that are created by default through the console are:

- username
- password
- engine
- host
- port
- dbname
- dbInstanceIdentifier

When replicating the same approach via Terraform only the engine cannot be sourced using outputs.
```
resource "random_password" "master" {
  length           = 16
  special          = true
  override_special = "_!%^"
}

resource "aws_secretsmanager_secret" "rds_credentials" {
  name = "rds-credentials"
}

resource "aws_secretsmanager_secret_version" "rds_credentials" {
  secret_id     = aws_secretsmanager_secret.rds_credentials.id
  secret_string = jsonencode(
    {
      "username": "${module.db.db_instance_username}",
      "password": "${random_password.master.result}",
      "engine": "postgres",
      "host": "${module.db.db_instance_address}",
      "port": "${module.db.db_instance_port}",
      "dbname": "${module.db.db_instance_name}"
      "dbInstanceIdentifier": "${module.db.db_instance_id}"
    }
  )
}
```

## Breaking Changes

No.

## How Has This Been Tested?
- [ x ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ x ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ x ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Review by antonbabenko — COMMENTED** [review]
LGTM. Could you also add one more related output - [`engine_version_actual`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#engine_version_actual) ?

**Comment by stefan-matic** [comment]
@antonbabenko Added the `engine_version_actual` as output `db_instance_engine_version`

File: modules/db_instance/outputs.tf [hunk 1]
```diff
@@ -28,6 +28,16 @@ output "db_instance_endpoint" {
   value       = try(aws_db_instance.this[0].endpoint, "")
 }
 
+output "db_instance_engine" {
+  description = "The RDS engine type"
+  value       = try(aws_db_instance.this[0].engine, "")
+}
+
+output "db_instance_engine_version" {
```
Comment (bryantbiggs) [inline modules/db_instance/outputs.tf:36]: we should carry through the actual attribute name like
```suggestion
output "db_instance_engine_version_actual" {
```

File: modules/db_instance/outputs.tf [hunk 2]
```diff
@@ -28,6 +28,16 @@ output "db_instance_endpoint" {
   value       = try(aws_db_instance.this[0].endpoint, "")
 }
 
+output "db_instance_engine" {
+  description = "The RDS engine type"
```
Comment (bryantbiggs) [inline modules/db_instance/outputs.tf:32]: from the provider docs
```suggestion
  description = "The database engine"
```

File: modules/db_instance/outputs.tf [hunk 3]
```diff
@@ -28,6 +28,16 @@ output "db_instance_endpoint" {
...
+  value       = try(aws_db_instance.this[0].engine, "")
+}
+
+output "db_instance_engine_version" {
+  description = "The RDS engine version"
```
[the remark this answers is not in the document]
Comment (stefan-matic) [reply modules/db_instance/outputs.tf]: Resolved in [df19752](https://github.com/terraform-aws-modules/terraform-aws-rds/pull/404/commits/df19752ae63c98efabba102c51f40f5675ea78d0)

**Review by bryantbiggs — APPROVED** [review]
Thanks for the PR @stefan-matic - looks great

**Comment by antonbabenko** [comment]
This PR is included in version 4.4.0 :tada:

**Pull request merged**